client/feature_cache.go

```diff
@@ -45,3 +45,10 @@ func (fc *FeatureCache) Get(name string) *Feature {
 
 	return feature
 }
+
+func (fc *FeatureCache) GetAll() map[string]*Feature {
+	fc.lock.RLock()
+	defer fc.lock.RUnlock()
+
+	return fc.features
```

Comment:
Map are passed by reference. That means that whoever gets this can modify the contents of the cache without any locks. Instead, deep-copy the map and return the copy. The features itself are also pointers and each contain a map inside, so you can not just loop over the key,values and put them in a new map. 

The easiest way to really deep-copying it can maybe be just marshall and unmarsall it into something (JSON?)  ¯\_(ツ)_/¯